keps/sig-scheduling/5234-dra-resourceslice-mixins/README.md

```diff
@@ -0,0 +1,902 @@
+  // to extend the set of attributes and capacities it already defines.
+  //
+  // The maximum number of attributes, capacity, and counters across all
+  // mixins is 256.
+  //
+  // +optional
+  // +listType=atomic
+  Device []DeviceMixin
+
+  // DeviceCounterConsumption represents a list of counter
+  // consumption mixins, each of which contains a set of counters
+  // that a device will consume from a counter set.
+  //
+  // The maximum number of attributes, capacity, and counters across all
+  // mixins is 256.
+  //
+  // +optional
+  // +listType=atomic
+  DeviceCounterConsumption []DeviceCounterConsumptionMixin
+
+  // CounterSet represents a list of counter set mixins, i.e.
+  // a collection of counters that a CounterSet can "include"
+  // to extend the set of counters it already defines.
+  //
+  // The maximum number of attributes, capacity, and counters across all
+  // mixins is 256.
+  //
+  // +optional
+  // +listType=atomic
+  CounterSet []CounterSetMixin
+}
+
+// DeviceMixin defines a mixin that can be referenced from a device.
+type DeviceMixin struct {
+  // Name is a unique identifier among all device mixins in the ResourceSlice.
+  // It must be a DNS label.
+  //
+  // +required
+  Name string
+
+  // Attributes defines the set of attributes for this mixin.
+  // The name of each attribute must be unique in that set.
+  //
+  // To ensure this uniqueness, attributes defined by the vendor
+  // must be listed without the driver name as domain prefix in
+  // their name. All others must be listed with their domain prefix.
+  //
+  // The maximum number of attributes, capacity, and counters across all
+  // mixins is 256.
+  //
+  // +optional
+  Attributes map[QualifiedName]DeviceAttribute
+
+  // Capacity defines the set of capacities for this mixin.
+  // The name of each capacity must be unique in that set.
+  //
+  // To ensure this uniqueness, capacities defined by the vendor
+  // must be listed without the driver name as domain prefix in
+  // their name. All others must be listed with their domain prefix.
+  //
+  // The maximum number of attributes, capacity, and counters across all
+  // mixins is 256.
+  //
+  // +optional
+  Capacity map[QualifiedName]DeviceCapacity
+}
+
+// DeviceCounterConsumptionMixin defines a mixin that
+// devices can include to extend or override the set of counters
+// that a device consumes from a counter set.
+type DeviceCounterConsumptionMixin struct {
+  // Name is a unique identifier among all device counter consumption
+  // mixins in the ResourceSlice. It must be a DNS label.
+  //
+  // +required
+  Name string
+
+  // Counters defines a set of counters
+  // that a device will consume from a counter set.
+  //
+  // The maximum number of attributes, capacity, and counters across all
+  // mixins is 256.
+  //
+  // +required
+  Counters map[string]Counter
+}
+
+// CounterSetMixin defines a mixin that a capacity pool can include.
+type CounterSetMixin struct {
```

Comment:
nit: can we move this up so it immediately follows `ResourceSliceMixins` (where there is a reference to this new type)

Reply:
I see that `ResourceSliceMixins` includes three new types and they are in that order, ignore!